### DISCUSSION TITLE
AHK "Pause" in python scripts?

### DISCUSSION THREAD

**Discussion by 0mn1try** [spyoungtech/ahk#224] [Q&A]
Hi, i'm kinda new to  programming, have about a year experience with AHK, in my application field some scripts that activate by a hotkey take decent time to execute(it could be up to 5 mins) and i like that in AHK i have separate hotkey for Pausing/Unpausing execution of the whole script at any time(usually if smth goes wrong ofc). Now i'm trying to switch to Python, and i can't find any way to do similar functionality of pausing/unpausing in Python. I was very excited to find out this ahk-python project yesterday, and when i looked in your documentation - there was some about suspend/ressume, but this is not the same as Pause/Unpause. 

So my question is - any chance i can implement smth similar to Pause/Unpause for the whole script written with Python-ahk like it was in pure AHK?
    
PS this project looks awesome, i like it <3

**Comment by MahsaBakhtiari** [comment]
hey @0mn1try! Yes, you can implement a similar Pause/Unpause functionality for your Python scripts using the ahk-python library. it's so cool to find others that are having fun with their projects! keep up the good work.

**Reply by 0mn1try** [reply]
yeah, cool, ok. But how exactly should i do that? I stopped digging and went here to ask when i got to threading in python, but couldn't find any way to stop another thread from outside, without inserting "if" checks inside function i want to stop. In AHK i just call "Pause" command from some hotkey in any time while another hotkeys' functionality is running, and that stops the whole thing, i don't have to insert flag-checks inside functionality that i want to freeze.
